include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
Or, to rephrase, if we continue seeing UB being reported after changing the code to `if (n != 0) memcpy(...)`, it would mean that we have another issue, since we should never be calling with `n` set to non-zero and `src` set to `NULL`.

I would like to see such case being reported. If we implement the check as `if (src != NULL) memcpy(...)`, we will lose the chance of receiving a report for such case.